ts/data/test.ts

```diff
@@ -0,0 +1,371 @@
+          samplePhasePendingWork(testTableView.test.libraryPreparations)
+        ) {
+          if (testTableView.test.libraryPreparations.length) {
+            addSpace(fragment);
+          }
+          addConstructionIcon("library preparation", fragment);
+        }
+      },
+      getCellHighlight(testTableView) {
+        testTableView.test = assertNotNull(testTableView.test);
+        if (
+          !testTableView.test.libraryPreparations.length &&
+          testTableView.test.libraryPreparationSkipped
+        ) {
+          return "na";
+        }
+        return getSamplePhaseHighlight(
+          testTableView,
+          testTableView.test.libraryPreparations
+        );
+      },
+    },
+    {
+      title: "Library Qualification",
+      addParentContents(testTableView, fragment) {
+        if (
+          handleNaSamplePhase(
+            testTableView,
+            testTableView.test.libraryQualifications,
+            fragment
+          )
+        ) {
+          return;
+        }
+        addSampleIcons(
+          testTableView,
+          testTableView.test.libraryQualifications,
+          fragment
+        );
+        if (
+          samplePhaseComplete(testTableView.test.libraryPreparations) &&
+          samplePhasePendingWork(testTableView.test.libraryQualifications)
+        ) {
+          if (testTableView.test.libraryQualifications.length) {
+            addSpace(fragment);
+          }
+          addConstructionIcon("library qualification", fragment);
+        }
+      },
+      getCellHighlight(testTableView) {
+        testTableView.test = assertNotNull(testTableView.test);
+        return getSamplePhaseHighlight(
+          testTableView,
+          testTableView.test.libraryQualifications
+        );
+      },
+    },
+    {
+      title: "Full-Depth Sequencing",
+      addParentContents(testTableView, fragment) {
+        if (
+          handleNaSamplePhase(
+            testTableView,
+            testTableView.test.fullDepthSequencings,
+            fragment
+          )
+        ) {
+          return;
+        }
+        addSampleIcons(
+          testTableView,
+          testTableView.test.fullDepthSequencings,
+          fragment
+        );
+        if (
+          samplePhaseComplete(testTableView.test.libraryQualifications) &&
+          samplePhasePendingWork(testTableView.test.fullDepthSequencings)
+        ) {
+          if (testTableView.test.fullDepthSequencings.length) {
+            addSpace(fragment);
+          }
+          addConstructionIcon("full-depth sequencing", fragment);
+        }
+      },
+      getCellHighlight(testTableView) {
+        testTableView.test = assertNotNull(testTableView.test);
+        return getSamplePhaseHighlight(
+          testTableView,
+          testTableView.test.fullDepthSequencings
+        );
+      },
+    },
+  ],
+};
+
+function handleNaSamplePhase(
+  testTableView: TestTableView,
+  samples: Sample[],
+  fragment: DocumentFragment
+) {
```

Comment:
Only difference from the function in `case.ts` is it takes a `Case` instead of `TestTableView`, but both only really need the `Requisition`. Change the original function to take a requisition instead and use that in both places.

Similar situation for `addSampleIcons` and `getSamplePhaseHighlight` below